corehq/apps/userreports/reports/builder/forms.py

```diff
@@ -849,19 +839,21 @@ def data_source_name(self):
 class RegistryCaseDataSourceHelper(CaseDataSourceHelper):
     def __init__(self, domain, registry_slug, source_type, source_id):
         assert source_type == 'case'
-
-        self.domain = domain
         self.registry_slug = registry_slug
-        self.source_type = source_type
-        self.source_id = source_id
+        super().__init__(domain, source_type, source_id)
 
-        owning_domain = DataRegistry.objects.get(slug=self.registry_slug).domain
+        registry = DataRegistry.objects.get(slug=self.registry_slug)
+        assert domain in registry.visible_domains()
+        DataRegistryHelper(self.domain, registry=registry).pre_access_check(source_type)
+
+        owning_domain = registry.domain
```

Comment:
it's easier if you let the helper load the registry for you as well since it will also do the access check then
```suggestion
        helper = DataRegistryHelper(self.domain, registry_slug=self.registry_slug)
        helper.pre_access_check(source_type)
        owning_domain = helper.registry.domain
```